app/assets/javascripts/blacklight_oembed/oembed.js

```diff
@@ -0,0 +1,31 @@
+export default function oEmbed(elements) {
+  if (!elements) return;
+
+  // Ensure elements is an array-like collection
+  elements = elements instanceof NodeList ? Array.from(elements) : [elements];
+
+  elements.forEach(function(embedViewer) {
+    const embedURL = embedViewer.dataset.embedUrl; // Get the embed URL from the data attribute
```

Comment:
Should you use safe access? `embedViewer?.dataset?.embedUrl`
Not sure of the case where there is not dataset but it seems possible.

Reply:
A HTMLElement will always return a dataset https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset